website/source/docs/secrets/rabbitmq/index.html.md

```diff
@@ -0,0 +1,326 @@
+and password "password" connecting to a local RabbitMQ management instance. It is important
+that the Vault user have the administrator privilege to manager users.
+
+Optionally, we can configure the lease settings for credentials generated
+by Vault. This is done by writing to the `config/lease` key:
+
+```
+$ vault write postgresql/config/lease lease=1h lease_max=24h
+Success! Data written to: postgresql/config/lease
+```
+
+This restricts each credential to being valid or leased for 1 hour
+at a time, with a maximum use period of 24 hours. This forces an
+application to renew their credentials at least hourly, and to recycle
+them once per day.
+
+The next step is to configure a role. A role is a logical name that maps
+to tags and virtual host permissions used to generated those credentials. For example,
+lets create a "readwrite" virtual host role:
+
+```text
+$ vault write rabbitmq/roles/readwrite \
+    vhosts='{"/":{"write": ".*", "read": ".*"}}'
+Success! Data written to: rabbitmq/roles/readonly
+```
+
+By writing to the `roles/readwrite` path we are defining the `readwrite` role.
+This role will be created by evaluating the given `vhosts` and `tags` statements.
+By default, no tags and no virtual hosts are assigned to a role. You can read more
+about RabbitMQ management tags [here](https://www.rabbitmq.com/management.html#permissions).
+Configure, write, and read permissions are granted per virtual host.
+
+To generate a new set of credentials, we simply read from that role:
+Vault is now configured to create and manage credentials for RabbitMQ!
+
+```text
+$ vault read rabbitmq/creds/readwrite
+lease_id       rabbitmq/creds/readwrite/2740df96-d1c2-7140-c406-77a137fa3ecf
+lease_duration 3600
+lease_renewable	true
+password       e1b6c159-ca63-4c6a-3886-6639eae06c30
+username       root-4b95bf47-281d-dcb5-8a60-9594f8056092
+```
+
+By reading from the `creds/readwrite` path, Vault has generated a new
+set of credentials using the `readwrite` role configuration. Here we
+see the dynamically generated username and password, along with a one
+hour lease.
+
+Using ACLs, it is possible to restrict using the rabbitmq backend such
+that trusted operators can manage the role definitions, and both
+users and applications are restricted in the credentials they are
+allowed to read.
+
+If you get stuck at any time, simply run `vault path-help rabbitmq` or with a
+subpath for interactive help output.
+
+## API
+
+### /rabbitmq/config/connection
+#### POST
+
+<dl class="api">
+  <dt>Description</dt>
+  <dd>
+    Configures the connection string used to communicate with RabbitMQ.
+    This is a root protected endpoint.
+  </dd>
+
+  <dt>Method</dt>
+  <dd>POST</dd>
+
+  <dt>URL</dt>
+  <dd>`/RabbitMQ/config/connection`</dd>
+
+  <dt>Parameters</dt>
+  <dd>
+    <ul>
+      <li>
+        <span class="param">uri</span>
+        <span class="param-flags">required</span>
+        The RabbitMQ management connection URI. e.g. "http://localhost:15672"
```

Comment:
Since the example is mentioned in the texts above, there might not be a need to mention it in the API section.